scripts/generate_upload_requirements_layer.sh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bash
+# This script is used to generate a dependency layer for the lambdas to utilize
+# It's not really for direct usage, `make generate-dependencies` can be used
+# but its part of the `make deploy` 
+
+set -euo pipefail
+shopt -s nullglob;
+
+account_id=$(aws sts get-caller-identity | jq -r ".Account" )
+populated_dss_tf_backend=$(echo ${DSS_TERRAFORM_BACKEND_BUCKET_TEMPLATE} | sed 's/{account_id}/'"${account_id}"'/'  )
+
+if [[ $(aws s3api get-bucket-location --bucket ${populated_dss_tf_backend} &>/dev/null ;echo $?) -ne 0 ]]
+then
+    echo "verify if bucket: ${populated_dss_tf_backend} exist"
+    exit 1
+fi
+
+build_path="$DSS_HOME/dependencies/python/lib/python3.6/site-packages"
+dependency_dir="$DSS_HOME/dependencies"
+aws_req_key=$DSS_DEPLOYMENT_STAGE/requirements.txt
+aws_zip_key=$DSS_DEPLOYMENT_STAGE/dss-dependencies-$DSS_DEPLOYMENT_STAGE.zip
+local_temp_chalice=${DSS_HOME}/temp_chalice
+local_req=$DSS_HOME/requirements.txt
+local_zip=$DSS_HOME/dss-dependencies-$DSS_DEPLOYMENT_STAGE.zip
+layer_name=dss-dependencies-${DSS_DEPLOYMENT_STAGE}
+
+function create_layer(){
+    layer_version_arn=$(aws lambda publish-layer-version --layer-name $layer_name --content S3Bucket=${populated_dss_tf_backend},S3Key=${aws_zip_key}| jq -r .LayerVersionArn )
+    echo "created layer/-version_arn: ${layer_version_arn}"
+}
+
+
+function build_clean_zip(){
+	mkdir -p $build_path
+	unzip -q -o -d $build_path deployment.zip
+	cd $dependency_dir
+	zip -qq -r -o $local_zip ./
+    cd ..
+}
+
+function build_chalice(){
+	# utilize challice to build out dependencies
+	# chalice requires a a configured chalice application to package.
+	# copy chalice folder TODO verify that in process this folder will be clean....
+	cp -R ${DSS_HOME}/chalice ${local_temp_chalice}
```

Comment:
Is there a reason to create a temporary directory? What's wrong building chalice in place?

Reply:
We have removed vendor.in/requirements.txt from the chalice/daemons folder, we could move them into the folder during this build_chalice; but in the event that something errors out the state of the chalice folder could be mutated, by performing this copy we can have a clean "environment" to perform the packaging, then clean it up when we are done, or just delete the folder as needed if something fails.

Reply:
Ok, but note that `chalice/.chalice/config.json` is mutated during build.